src/requests/typings.ts

```diff
@@ -74,4 +88,11 @@ export type NormalizedCanonicalRequest = {
   body: CanonicalRequest['body']
 }
 
-export type SignedRequestHeaders = { [key in ContentfulHeader]: string }
+export type Subject = XOR<ContentfulHeaderWithApp, ContentfulHeaderWithUser>
+
+export type ContextHeaders = {
```

Comment:
I think this type is incorrect actually. As it doesn't allow the possibility of not having a subject - which is a case we want to allow.

Currently if you try

```
      signRequest(VALID_SECRET, VALID_REQUEST, VALID_TIMESTAMP, {
        'x-contentful-space-id': 'whatever',
        'x-contentful-environment-id': 'whatever',
      })
```

we get a typescript error